### PR TITLE
Write lesson on variables in memory and using the visualizer

### PR DESCRIPTION
**Pull request by kcleland0818**

### CODE REVIEW

**Review by linda-seiter — COMMENTED** [review]
"Comprehension check! What do we think this program would output if we were to run it?"
this paragraph seems to appear twice.

"run this program in debug"   this phrase seems a bit off.  maybe "debug mode"?

"line number in the editor window - a red dot will appear to indicate that you successfully set a breakpoint on that line:"
 tell them to click again on the breakpoint to remove it if the set it on the wrong line?

typo re. "Java Visualizer tab that re highlighted"

"We could also copy and paste this code into the Java Visualizer."  
this seems confusing because they just used the Java Visualizer plugin.  
Maybe "browser based Java Visualizer" or "Java Visualizer at pythontutor.com".


"Let's put a breakpoint on line 9 and run the program in the debugger to see what happens with memory when we re-assign the variables x and z:"

tell them to remove the existing breakpoint at line 5.

**Review by linda-seiter — APPROVED** [review]
Looks good!

File: README.md [hunk 1]
```diff
@@ -2,31 +2,177 @@
...
+
+[Java Visualizer for Variables and Call Stack Lesson](https://pythontutor.com/visualize.html#code=public%20class%20Main%20%7B%0A%20%20%20%20public%20static%20void%20main%28String%5B%5D%20args%29%20%7B%0A%20%20%20%20%20%20%20%20int%20x%20%3D%201%3B%0A%20%20%20%20%20%20%20%20int%20y%20%3D%202%3B%0A%20%20%20%20%20%20%20%20int%20z%20%3D%203%3B%0A%20%20%20%20%7D%0A%7D&cumulative=false&curInstr=1&heapPrimitives=nevernest&mode=display&origin=opt-frontend.js&py=java&rawInputLstJSON=%5B%5D&textReferences=false)
+
+For convenience, we have also embedded the Java Visualizer you would see on a
+browser here:
+
+<iframe width="800" height="500" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=public%20class%20Main%20%7B%0A%20%20%20%20public%20static%20void%20main%28String%5B%5D%20args%29%20%7B%0A%20%20%20%20%20%20%20%20int%20x%20%3D%201%3B%0A%20%20%20%20%20%20%20%20int%20y%20%3D%202%3B%0A%20%20%20%20%20%20%20%20int%20z%20%3D%203%3B%0A%20%20%20%20%7D%0A%7D&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=1&heapPrimitives=nevernest&origin=opt-frontend.js&py=java&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
```
Comment (developerjaym) [inline README.md:118]: I'm not sure this embedded properly. I see this HTML on the markdown preview

**Review by developerjaym — CHANGES_REQUESTED** [review]
Looked like the iframe didn't work properly

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md:118]: Github won't display it correctly, but canvas will as long as IntelliJ previewer does.

Comment (kcleland0818) [reply README.md:118]: @developerjaym - see Linda's comment above. It doesn't show correctly in IntelliJ, but it does show a big blank space of where it should be embedded. Linda has tested the process in the Canvas page I shared on the se-java-team slack channel.

**Review by developerjaym — APPROVED** [review]

**Pull request merged**